src/node_crypto.cc

```diff
@@ -5354,6 +5354,25 @@ void GetCurves(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void AddEntropy(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (args.Length() == 0) {
+    // Delegate entropy generation to OpenSSL, which will add
+    // entropy from system sources
+    RAND_poll();
+    return;
+  }
+  // Make sure we got a buffer from the user and use it to
+  // seed OpenSSL.
+  THROW_AND_RETURN_IF_NOT_BUFFER(args[0]);
+  Local<Object> bufObj = args[0]->ToObject();
+  const void* buf = Buffer::Data(bufObj);
+  size_t bufLength = Buffer::Length(bufObj);
+  RAND_seed(buf, bufLength);
```

Comment:
I don't have a use case in mind for wanting to supply my own entropy generated outside of the OS, so no, I think we could just call RAND_poll() and leave out the call to RAND_add() entirely.